vim-patch:8.0.{1082,1497,1498,1513},8.1.{901,1357,1358}

janlazo opened the pull request:
**vim-patch:8.0.1082: tests fail when run under valgrind**
Problem:    Tests fail when run under valgrind.
Solution:   Increase waiting times.
vim/vim@9d18961

**vim-patch:8.0.1497: getting the jump list requires parsing the output of :jumps**
Problem:    Getting the jump list requires parsing the output of :jumps.
Solution:   Add getjumplist(). (Yegappan Lakshmanan, closes vim/vim#2609)
vim/vim@4f50588

**vim-patch:8.0.1498: getjumplist() returns duplicate entries**
Problem:    Getjumplist() returns duplicate entries. (lacygoill)
Solution:   Call cleanup_jumplist(). (Yegappan Lakshmanan)
vim/vim@a7e18d2

**vim-patch:8.0.1513: the jumplist is not always properly cleaned up**
Problem:    The jumplist is not always properly cleaned up.
Solution:   Call fname2fnum() before cleanup_jumplist(). (Yegappan Lakshmanan)
vim/vim@4867974

**vim-patch:8.1.0901: index in getjumplist() may be wrong**
Problem:    Index in getjumplist() may be wrong. (Epheien)
Solution:   Call cleanup_jumplist() earlier. (Yegappan Lakshmanan, closes vim/vim#3941)
vim/vim@57ee2b6

**vim-patch:8.1.1357: test 37 is old style**
Problem:    Test 37 is old style.
Solution:   Turn it into a new style test. (Yegappan Lakshmanan, closes vim/vim#4398)
vim/vim@999dc14

**vim-patch:8.1.1358: cannot enter character with a CSI byte**
Problem:    Cannot enter character with a CSI byte.
Solution:   Only check "gui.in_use" when VIMDLL is defined. (Ken Takata, closes vim/vim#4396)
vim/vim@386b43e

janlazo committed:
vim-patch:8.0.1082: tests fail when run under valgrind

Problem:    Tests fail when run under valgrind.
Solution:   Increase waiting times.
https://github.com/vim/vim/commit/9d18961323a2a5c3b609c98ce0d78613c71f3532

janlazo committed:
vim-patch:8.0.1497: getting the jump list requires parsing the output of :jumps

Problem:    Getting the jump list requires parsing the output of :jumps.
Solution:   Add getjumplist(). (Yegappan Lakshmanan, closes vim/vim#2609)
https://github.com/vim/vim/commit/4f50588ba336e7f086a72c53f5688c2494fc34b3

janlazo committed:
vim-patch:8.0.1498: getjumplist() returns duplicate entries

Problem:    Getjumplist() returns duplicate entries. (lacygoill)
Solution:   Call cleanup_jumplist(). (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/a7e18d237f817637815f0de44b08df1e0ca0f4f9

janlazo committed:
vim-patch:8.0.1513: the jumplist is not always properly cleaned up

Problem:    The jumplist is not always properly cleaned up.
Solution:   Call fname2fnum() before cleanup_jumplist(). (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/486797413791f6be12dcec6e5faf4f952e4647ae

janlazo committed:
vim-patch:8.1.0901: index in getjumplist() may be wrong

Problem:    Index in getjumplist() may be wrong. (Epheien)
Solution:   Call cleanup_jumplist() earlier. (Yegappan Lakshmanan,
            closes vim/vim#3941)
https://github.com/vim/vim/commit/57ee2b6e0b5b730d12ee9db00a8e2a577df9e374

janlazo committed:
lint

justinmk commented:
```
../src/nvim/mark.c:1189:9: error: use of undeclared identifier 'i'
    if (i >= wp->w_jumplistlen) {  // no duplicate
```

janlazo committed:
vim-patch:8.1.1357: test 37 is old style

Problem:    Test 37 is old style.
Solution:   Turn it into a new style test. (Yegappan Lakshmanan, closes vim/vim#4398)
https://github.com/vim/vim/commit/999dc14644b8a9530ce0da22e90ca402c95c9c5a

janlazo committed:
vim-patch:8.1.1358: cannot enter character with a CSI byte

Problem:    Cannot enter character with a CSI byte.
Solution:   Only check "gui.in_use" when VIMDLL is defined. (Ken Takata,
            closes vim/vim#4396)
https://github.com/vim/vim/commit/386b43e59498cc7b52a60f09f74bdb44df99386c
